script.js

```diff
@@ -1,4 +1,78 @@
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var gameMode = "waitingForPlayerName";
+var playerNames = [];
+var playerOneScore = [];
+var playerTwoScore = [];
+var playerOneScoreArray = [];
+var playerTwoScoreArray = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (gameMode == "waitingForPlayerName") {
+    //Player One Code
+    if (input == "") {
+      return `Please Input Player 1 Name`;
+    } else {
+      playerNames.push(input);
+      gameMode = "rollDicePlayerOne";
+      return `Hello ${playerNames[0]}, welcome to Beat That! Click on Submit to Roll the dice!`;
+    }
+  }
+  if (gameMode == "rollDicePlayerOne") {
+    playerOneScoreArray.push(rollDice());
+    playerOneScoreArray.push(rollDice());
```

Comment:
This could be a helper function outside of the main function which can then be reused for lines 53-54.